File: Atividades_java/Media_media_vetor_melhorado.java
package estudos.atividade;
import javax.swing.JOptionPane;
public class Media_media_vetor_melhorado {

	public static void main(String[] args) {
		 					
				float nota[] = new float[5];
				float calc_media,soma_notas;
				int contador;
				String msg="";
				contador = 0;
				soma_notas = 0;
				for (int i=0;i<=4;i++)
				{
				nota[i]	= Float.parseFloat(JOptionPane.showInputDialog("Digite a "+i+". nota: "));
				soma_notas = soma_notas + nota[i];	
				}
				calc_media = soma_notas / 5;
				for (int i = 0;i<=4;i++)
					if (nota[i] >= calc_media)
					contador = contador + 1;
				msg = msg + "A média é "+calc_media+"\n"+contador+" Aluno(s) ficaram com a nota acima da média";
				JOptionPane.showMessageDialog(null, msg);
				System.out.println(msg);

	}

}

File: Arquivos_fonte_C_2/Media_media_vetor_melhorado.cpp
#include <conio.h>
#include <stdio.h>

main()
{
	float nota[5];
 	float calc_media,soma_notas;
	int i,contador;
	contador = 0;
	soma_notas = 0; 
	for (i=0;i<=4;i++)
	{
	printf("Digite a %d. nota = ",i+1);
	scanf("%f",&nota[i]);
	soma_notas = soma_notas + nota[i];
  	}
  	calc_media = soma_notas / 5;
	for (i=0;i<=4;i++)
	{
		if (nota[i] >= calc_media)
		contador++;
	}
	printf("A media e %f",calc_media);
	printf("\n%d Aluno(s) ficaram com a nota acima da media",contador);
	getch();
}
